crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -154,50 +169,71 @@ impl BlobVec {
     ///
     /// # Safety
     /// It is the caller's responsibility to ensure that `index` is < `self.len()`
-    /// Callers should _only_ access the returned pointer immediately after calling this function.
     #[inline]
-    pub unsafe fn swap_remove_and_forget_unchecked(&mut self, index: usize) -> *mut u8 {
+    pub unsafe fn swap_remove_and_forget_unchecked(&mut self, index: usize) -> OwningPtr<'_> {
         debug_assert!(index < self.len());
         let last = self.len - 1;
         let swap_scratch = self.swap_scratch.as_ptr();
         std::ptr::copy_nonoverlapping(
-            self.get_unchecked(index),
```

Comment:
Not huge priority but it feels like `OwningPtr::as_ptr()` might make these easier to read